From 3894123503c4b9c237dfcb13e4fd45fcd0e56019 Mon Sep 17 00:00:00 2001
From: "Daniel P. Berrange" <berrange@redhat.com>
Date: Tue, 13 Aug 2013 17:33:22 +0100
Subject: [PATCH] Fix upgrade command wrt to generic containers

For https://bugzilla.redhat.com/show_bug.cgi?id=994371

The generic containers do not include any integration with
systemd on the host. So during upgrade we must skip the
unit file rewriting

Signed-off-by: Daniel P. Berrange <berrange@redhat.com>
(cherry picked from commit 85b429cca26203c61e48c327139e36721c62c741)
---
 bin/virt-sandbox-service | 36 ++++++++++++++++++------------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/bin/virt-sandbox-service b/bin/virt-sandbox-service
index 03873c9..3e83c94 100755
--- a/bin/virt-sandbox-service
+++ b/bin/virt-sandbox-service
@@ -928,28 +928,28 @@ def upgrade_config_legacy(path):
     else:
         container = SystemdContainer(uri=args.uri, config=config)
 
-    fd = open(container.get_unit_path())
-    unitfile = fd.read()
-    fd.close()
+        fd = open(container.get_unit_path())
+        unitfile = fd.read()
+        fd.close()
 
-    unitfile = unitfile.replace("/usr/bin/virt-sandbox-service start",
-                                "/usr/libexec/virt-sandbox-service-util -c lxc:/// -s")
-    unitfile = unitfile.replace("/usr/bin/virt-sandbox-service reload",
-                                "/usr/bin/virt-sandbox-service -c lxc:/// reload")
-    unitfile = unitfile.replace("/usr/bin/virt-sandbox-service stop",
-                                "/usr/bin/virsh -c lxc:/// destroy")
+        unitfile = unitfile.replace("/usr/bin/virt-sandbox-service start",
+                                    "/usr/libexec/virt-sandbox-service-util -c lxc:/// -s")
+        unitfile = unitfile.replace("/usr/bin/virt-sandbox-service reload",
+                                    "/usr/bin/virt-sandbox-service -c lxc:/// reload")
+        unitfile = unitfile.replace("/usr/bin/virt-sandbox-service stop",
+                                    "/usr/bin/virsh -c lxc:/// destroy")
 
-    unitfile = re.sub("WantedBy=.*\.target",
-                      "WantedBy=multi-user.target",
-                      unitfile)
+        unitfile = re.sub("WantedBy=.*\.target",
+                          "WantedBy=multi-user.target",
+                          unitfile)
 
-    os.remove(container.get_unit_path())
-    fd = open(container.get_unit_path(), "wx")
-    fd.write(unitfile)
-    fd.close()
+        os.remove(container.get_unit_path())
+        fd = open(container.get_unit_path(), "wx")
+        fd.write(unitfile)
+        fd.close()
 
-    sys.stdout.write(_("Created unit file %s\n") %
-                     container.get_unit_path())
+        sys.stdout.write(_("Created unit file %s\n") %
+                         container.get_unit_path())
 
     # Create new config file + libvirt persistent XML config
     container.save_config()
